src/main/scala/com/actian/spark_vector/colbuffer/util/TimeConversion.scala

```diff
@@ -39,27 +39,25 @@ object TimeConversion {
 
   @inline final def unscaleNanos(scaledNanos: Long, scale: Int): Long = scaledNanos * PowersOfTen(NanosecondsScale - scale)
 
-  @inline private final def convertLocalDateHelper(date: Date, sign: Int = 1): Unit = {
-    val cal = Calendar.getInstance()
+  @inline private final def convertLocalDateHelper(date: Date, sign: Int = 1, cal: Calendar): Unit = {
     cal.setTime(date)
     date.setTime(date.getTime + sign * cal.get(Calendar.ZONE_OFFSET) + sign * cal.get(Calendar.DST_OFFSET))
   }
 
-  final def convertLocalDateToUTC(date: Date): Unit = convertLocalDateHelper(date)
+  final def convertLocalDateToUTC(date: Date, cal: Calendar): Unit = convertLocalDateHelper(date, 1, cal)
```

Comment:
For convenience, specify default value for `cal` and drop a comment mentioning that, for performance reasons, the caller may want to pass it explicitly.